src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs

```diff
@@ -46,23 +48,23 @@ public partial class WorkitemController
     [VersionedPartitionRoute(KnownRoutes.AddWorkitemInstancesRoute, Name = KnownRouteNames.PartitionedAddWorkitemInstance)]
     [VersionedRoute(KnownRoutes.AddWorkitemInstancesRoute, Name = KnownRouteNames.AddWorkitemInstance)]
     [AuditEventType(AuditEventSubType.AddWorkitem)]
-    public async Task<IActionResult> AddAsync()
+    public async Task<IActionResult> AddAsync([FromBody] IReadOnlyCollection<DicomDataset> dicomDatasets)
     {
         // The Workitem UID is passed as the name of the first query parameter 
         string workitemUid = HttpContext.Request.Query.Keys.FirstOrDefault();
 
-        return await PostAddAsync(workitemUid);
+        return await PostAddAsync(workitemUid, dicomDatasets);
     }
 
-    private async Task<IActionResult> PostAddAsync(string workitemInstanceUid)
+    private async Task<IActionResult> PostAddAsync(string workitemInstanceUid, IReadOnlyCollection<DicomDataset> dicomDatasets)
     {
-        long fileSize = Request.ContentLength ?? 0;
-        _logger.LogInformation("DICOM Web Add Workitem Transaction request received, with Workitem instance UID {WorkitemInstanceUid}, and file size of {FileSize} bytes",
+        long numberOfDatasets = dicomDatasets?.Count ?? 0;
+        _logger.LogInformation("DICOM Web Add Workitem Transaction request received, with Workitem instance UID {WorkitemInstanceUid}, and {NumberOfDatasets} DICOM dataset",
             workitemInstanceUid ?? string.Empty,
```

Comment:
No need to null-coalesce here. `null` is valid in format strings

Reply:
Done